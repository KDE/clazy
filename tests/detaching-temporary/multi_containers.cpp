#include <QtCore/QMap>
#include <QtCore/QHash>

QMap<int,int> getMap() { return {}; }
QMultiMap<int,int> getMultiMap() { return {}; }
void testDerivedClass()
{
    getMultiMap().begin(); // Warning
    getMultiMap().insert(1, 1); // Warning
}

void maps()
{
    QMap<int, QStringList> map;
    map.value(0).first(); // OK for Qt5, value() returns const T
    map[0].removeAll("asd"); // OK
    map.values().first(); // OK, QMap::values() isn't shared
    getMap().keyValueBegin(); // Warning
}
void test_ctor()
{
    QStringList().first();
    QByteArray key = "key";
    QByteArray(key + key).data(); // In Qt5, this is const
}
