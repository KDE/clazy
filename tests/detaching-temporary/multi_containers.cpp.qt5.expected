detaching-temporary/multi_containers.cpp:8:5: warning: Don't call QMap::begin() on temporary [-Wclazy-detaching-temporary]
detaching-temporary/multi_containers.cpp:9:5: warning: Modifying temporary container is pointless and it also detaches [-Wclazy-detaching-temporary]
detaching-temporary/multi_containers.cpp:18:5: warning: Don't call QMap::keyValueBegin() on temporary [-Wclazy-detaching-temporary]
